src/nbl/video/CVulkanPhysicalDevice.h

```diff
@@ -255,6 +262,11 @@ class CVulkanPhysicalDevice final : public IPhysicalDevice
                 m_properties.limits.filterMinmaxImageComponentMapping = samplerFilterMinmaxProperties.filterMinmaxImageComponentMapping;
             }
 
+            if(m_properties.apiVersion >= VK_MAKE_API_VERSION(0, 1, 2, 0))
+            {
+                m_properties.limits.framebufferIntegerColorSampleCounts = vulkan12Properties.framebufferIntegerColorSampleCounts;
+            }
```

Comment:
what happens if someone doesn't have VK 1.2 ?

can they not use MSAA integer attachments on frambuffers?

Reply:
The only place that reports this is `VkPhysicalDeviceVulkan12Properties` and it is `// Provided by VK_VERSION_1_2`

Reply:
yes I know, my question is, what happens when someone wants to use MSAA integer attachment in VK 1.1
A) they can't
B) there is no limit on the number of samples
C) the limit is the same as something else, and the reason VK 1.2 introduces a separate one for integers is because the new limit is higher

Reply:
From what I read in the spec the supported value should be at least `ESCF_1_BIT`
![image](https://user-images.githubusercontent.com/10983665/176896690-bfc43945-e7d0-4cc7-858a-c5492aa7c505.png)

So It's neither of those choices because it might be different for each integer format, best way is to query your integer format from physical device using `vkGetPhysicalDeviceImageFormatProperties` and get the sampleCounts 
https://www.khronos.org/registry/vulkan/specs/1.3-extensions/man/html/VkImageFormatProperties.html

So I suggest setting this to maximum possible value by default and let the user query from physical device for the specific format

https://www.khronos.org/registry/vulkan/specs/1.3-extensions/html/vkspec.html#features-supported-sample-counts

Reply:
whats the point of exposing this as a limit, if it can be different for every texture format and usage?

Reply:
then dont expose this at all and just have us rely on the format query API to get sample counts accurately